### PR TITLE
add circle-ci

### PR DESCRIPTION
**Pull request by BenTheElder**
definitely not right yet, but should be easier to test from here...

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+version: 2
+jobs:
+  build:
+    docker:
+    machine:
```
Comment (munnerz) [inline .circleci/config.yml]: Is 'docker:' correct? Should this be 'docker: {}'?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+      enabled: true
+    steps:
+    - checkout
+    - run: ./kind-smoketest.sh
```
Comment (munnerz) [inline .circleci/config.yml]: ... is this real yaml?! ¯\_(ツ)_/¯

File: .circleci/config.yml (hunk 1 above)
Comment (BenTheElder) [reply .circleci/config.yml]: stray key, fixed!

File: .circleci/config.yml (hunk 2 above)
Comment (BenTheElder) [reply .circleci/config.yml]: I think yaml allows lists of mixed types technically, this is circle-ci yaml anyhow 🙃

**Comment by BenTheElder** [comment]
this works now :-)

once we have another release tag out, we should switch this to use that by default probably, but as an example I think this suffices for now hopefully

**Pull request closed without merging**